internal/indexer/indexer_buffer.go

```diff
@@ -0,0 +1,75 @@
+package indexer
+
+import (
+	"sync"
+
+	set "github.com/deckarep/golang-set/v2"
+
+	"github.com/stellar/wallet-backend/internal/indexer/types"
+)
+
+func NewIndexerBuffer() IndexerBuffer {
+	return IndexerBuffer{
+		Participants:           set.NewSet[string](),
+		txByHash:               make(map[string]types.Transaction),
+		participantTxHashBimap: NewBiMap[string, string](),
+	}
+}
+
+type IndexerBuffer struct {
+	mu                     sync.RWMutex
+	Participants           set.Set[string]
+	txByHash               map[string]types.Transaction
+	participantTxHashBimap *BiMap[string, string]
+}
+
+func (b *IndexerBuffer) PushParticipantTransaction(participant string, transaction types.Transaction) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	b.txByHash[transaction.Hash] = transaction
+	b.Participants.Add(participant)
+
+	b.participantTxHashBimap.Add(participant, transaction.Hash)
+}
+
+func (b *IndexerBuffer) GetNumberOfTransactions() int {
+	b.mu.RLock()
+	defer b.mu.RUnlock()
+
+	return len(b.txByHash)
+}
+
+func (b *IndexerBuffer) GetParticipantTransactionHashes(participant string) set.Set[string] {
+	b.mu.RLock()
+	defer b.mu.RUnlock()
+
+	return b.participantTxHashBimap.GetForward(participant).Clone()
+}
+
+func (b *IndexerBuffer) GetParticipantTransactions(participant string) []types.Transaction {
+	b.mu.RLock()
+	defer b.mu.RUnlock()
+
+	txs := []types.Transaction{}
+	it := b.participantTxHashBimap.GetForward(participant).Iterator()
+	for txHash := range it.C {
+		txs = append(txs, b.txByHash[txHash])
+	}
+
+	return txs
+}
+
+func (b *IndexerBuffer) GetTransactionParticipants(transactionHash string) set.Set[string] {
```

Comment:
When would we need to get a transaction's participants from the buffer? Looking at the code, we do that only once using the participants processor and then filter txns by iterating over each participant. I am unable to think of the case where we want to use the backward map in the bimap? 🤔

Reply:
Great catch! I added it there on my first implementation but then I forgot to remove it when doing a refactor.

Now I'm using just a regular map and revoced all the bimap code:
- https://github.com/stellar/wallet-backend/pull/219/commits/ee8bbcff77800a7390b430ba6a3e232b2ef132a5
- https://github.com/stellar/wallet-backend/pull/219/commits/fbe5fc37963bba13751131e5c1fc4f2d562d9d56